pkg/config/handlerConfigurer.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/golang/protobuf/proto"
+
+	pbd "istio.io/api/mixer/v1/config/descriptor"
+	pb "istio.io/mixer/pkg/config/proto"
+	"istio.io/mixer/pkg/expr"
+	"istio.io/mixer/pkg/template"
+)
+
+type (
+	handlerConfigurer struct {
+		tmplRepo       template.Repository
+		typeChecker    expr.TypeChecker
+		attrDescFinder expr.AttributeDescriptorFinder
+	}
+	instancesByTemplate struct {
+		instancesNamesByTemplate map[string][]string
+	}
+)
+
+func (t *instancesByTemplate) insertInstance(tmplName string, instName string) {
+	instsPerTmpl, exists := t.instancesNamesByTemplate[tmplName]
+	if !exists {
+		t.instancesNamesByTemplate[tmplName] = make([]string, 0)
+	}
+
+	if !contains(instsPerTmpl, instName) {
+		t.instancesNamesByTemplate[tmplName] = append(t.instancesNamesByTemplate[tmplName], instName)
+	}
+}
+
+func newInstancesByTemplate() instancesByTemplate {
+	return instancesByTemplate{make(map[string][]string)}
+}
+
+// ConfigureHandlers identifies and invokes all the type configuration (per template) that needs
+// to be done on a handler.
+func ConfigureHandlers(actions []*pb.Action, constructors map[string]*pb.Constructor,
```

Comment:
hmm.. ideally I would prefer to use ConfigureHandler, because that is what this func does.. however we cannot since the package name is config.. so for now, shall we leave it as is ? If one of us come up with a name that makes total sense.. we can just do file rename..